front/app/modules/commercial/flag_inappropriate_content/admin/components/ActivityWarningsTab.tsx

```diff
@@ -15,22 +15,29 @@ declare module 'modules/commercial/moderation/admin/containers/index' {
 
 interface Props {
   onData: (data: InsertConfigurationOptions<ITabItem>) => void;
+  activeFlagsCount: number;
 }
 
 const ActivityWarningsTab = ({
```

Comment:
So this component doesn't really render anything, but it is just there to deal with updates? Would it make more sense to then turn it into a hook? Also, the name `ActivityWarningsTab` makes it seem to me like it would render a tab, so it confused me a little to see it not render anything

Reply:
This is a mechanism to render components from a module. The way to insert is to render it in an Outlet, so it has to be a component. So it is actually the ActivityTab (it's the data that's added to the other tabs and that's passed to the Tab component in the core). Or perhaps I'm not entirely following. How would you solve this with a hook?

Reply:
I understand it's confusing initially. It takes some getting used to. But once it is, it's pretty straight forward in fact.